### DISCUSSION TITLE
Is it possible to lazy load this plugin yet interact with nvim-lspconfig?

### DISCUSSION THREAD

**Discussion by glyh** [hrsh7th/nvim-cmp#688] [Q&A] [answered]
I notice that the init script has this line:
```lua
    local capabilities = require('cmp_nvim_lsp').update_capabilities(vim.lsp.protocol.make_client_capabilities())
```

Where the capabilities are used by another plugin, nvim-lspconfig. So to make this work, I had to load this before nvim-lspconfig, which is itself not lazy-loaded. 

However, I want to lazy-loaded this plugin to event InsertEnter. So, is there a workaround? Thanks a lot!

**Comment by dmitmel** [comment]
Is lazy loading nvim-cmp really worth it? You'll be entering the insert mode in 99% of editing sessions. I have also measured the execution times of the `plugin/cmp.lua` script, the `lua/cmp/init.lua` module, and the `cmp.setup` call, and they all take ~0.6, ~5 and ~0.2 milliseconds respectively.

**Reply by glyh** [reply]
While lazy loading a single plugin does not affect the performance much, it can improve when you have a bunch of them...

**Reply by smjonas** [reply]
I agree with @dmitmel. I don't  think you will gain much by lazy-loading cmp. Lazy loading makes more sense for plugins that are used rarely / at special events only (like for a specific filetype or command). Entering insert mode is not a special event.

**Comment by arsham** [comment, marked as answer]
You can try adding this to the `nvim-lspconfig` use block:

```lua
use {
    'neovim/nvim-lspconfig',
    after = { 'nvim-cmp' },
    event = { 'BufRead', 'BufNewFile', 'InsertEnter' },
}
```

**Reply by glyh** [reply]
Thanks a lot!
